daemon/src/auto_rollover.rs

```diff
@@ -127,36 +128,49 @@ where
 }
 
 #[xtra_productivity(message_impl = false)]
-impl<O, M> Actor<O, M>
+impl<O> Actor<O>
 where
     O: 'static,
-    M: 'static,
-    M: xtra::Handler<monitor::StartMonitoring>,
-    O: xtra::Handler<oracle::MonitorAttestation> + xtra::Handler<oracle::GetAnnouncement>,
+    O: xtra::Handler<oracle::GetAnnouncement>,
 {
-    async fn handle_rollover_completed(&mut self, _: RolloverCompleted) -> Result<()> {
-        // TODO: Implement this in terms of event sourcing
+    async fn handle_rollover_completed(
+        &mut self,
+        rollover_completed: RolloverCompleted,
+    ) -> Result<()> {
+        let id = rollover_completed.order_id();
+
+        let mut conn = self.db.acquire().await?;
+        let cfd = load_cfd(id, &mut conn).await?;
+
+        let event = cfd.roll_over(rollover_completed)?;
+        append_event(event.clone(), &mut conn).await?;
+
+        if let Err(e) = self
+            .process_manager_actor
+            .send(process_manager::Event::new(event.clone()))
+            .await?
+        {
+            tracing::error!("Sending event to process manager failed: {:#}", e);
```

Comment:
My current view is that the `ProcessManager` failing is a hard error and should always abort the current function. If the caller wants to only log that result, that is up to them!

Reply:
Yeah, I agree with you.